feg/gateway/services/n7_n40_proxy/n7/model_conversion.go

```diff
@@ -597,6 +691,61 @@ func newUsageMonitoringUpdateResponse(
 	}
 }
 
+func getUsageMonitorsProtoForUpdate(
+	updateCtx *SmPolicyUpdateReqCtx,
+	smPolicyDecision *n7_sbi.SmPolicyDecision,
+) []*protos.UsageMonitoringUpdateResponse {
+	monitors := []*protos.UsageMonitoringUpdateResponse{}
+	// Usage monitorting at pcc rule level
+	if smPolicyDecision.PccRules != nil {
+		for _, pccRule := range smPolicyDecision.PccRules.AdditionalProperties {
+			if pccRule.RefUmData == nil || len((*pccRule.RefUmData)[0]) == 0 {
+				continue
+			}
+			umResp := newUsageMonitoringUpdateResponseFromUpdateCtx(
+				updateCtx, smPolicyDecision, (*pccRule.RefUmData)[0], protos.MonitoringLevel_PCC_RULE_LEVEL)
+			if umResp == nil {
+				continue
+			}
+			monitors = append(monitors, umResp)
+		}
+	}
+	// Usage monitoring at session level
+	if smPolicyDecision.SessRules != nil {
+		for _, sessRule := range smPolicyDecision.SessRules.AdditionalProperties {
+			if sessRule.RefUmData == nil || len(*sessRule.RefUmData) == 0 {
+				continue
+			}
+			umResp := newUsageMonitoringUpdateResponseFromUpdateCtx(
+				updateCtx, smPolicyDecision, *sessRule.RefUmData, protos.MonitoringLevel_SESSION_LEVEL)
+			if umResp == nil {
+				continue
+			}
+			monitors = append(monitors, umResp)
+		}
+	}
+	return monitors
+}
+
+func newUsageMonitoringUpdateResponseFromUpdateCtx(
+	updateCtx *SmPolicyUpdateReqCtx,
+	smPolicyDecision *n7_sbi.SmPolicyDecision,
+	umId string,
+	monLevel protos.MonitoringLevel,
+) *protos.UsageMonitoringUpdateResponse {
+	umEntry, found := smPolicyDecision.UmDecs.Get(umId)
+	if !found {
+		return nil
+	}
+	return &protos.UsageMonitoringUpdateResponse{
+		Credit:    getUsageMonitoringCreditsProto(umId, &umEntry, monLevel),
+		SessionId: updateCtx.SessionId,
+		TgppCtx:   updateCtx.TgppCtx,
+		Sid:       updateCtx.IMSI,
+		Success:   true,
+	}
+}
```

Comment:
all these new functions have as argument the request and the response, which, in my opinion makes it a little bit weird to read (it make sense, but weird). One solution would be just pass `sessionID`, `tgppCtx` and `Sid` separateley. But that may create a big signature. We could create a new type to pass those arguments (which would be similar object than `updateCtx`). But none of those solutions seems better than what you have.

So lets add a doc string to all those function reminding that `SmPolicyUpdateReqCtx` is used just to fill some of the fields the response from PCF does not provide